File: 5 Kyu/Valid Parentheses/solution.go
package kata

func ValidParentheses(parens string) bool {
	n := 0

	for _, r := range parens {
		s := string(r)

		if s == "(" {
			n++
		} else if s == ")" {
			if n < 1 {
				return false
			}

			n--
		}
	}

	return n == 0
}

File: 5 Kyu/Valid Parentheses/solution.js
function validParentheses(parens) {
    let expected = 0;

    for (const paren of parens.split("")) {
        if (paren === "(") expected++;
        else if (paren === ")") {
            if (expected < 1) return false;
            expected--;
        };
    }

    return expected === 0;
}